app/herdbook.py

```diff
@@ -29,7 +34,38 @@ def after_request(response):
 
     return response
 
+@APP.route('/api/user')
+def get_user():
+    """
+    Returns information on the current logged in user.
+    """
+    user = session.get('user')
+    if user:
+        return user
+    return jsonify(db.User().frontend_data())
+
+@APP.route('/api/login', methods=['POST'])
+def login():
+    """
+    Parses a login form and sets session variables when logged in.
+    If login fails the system will default to an anonymous user.
+    """
 
+    form = request.json
+    # Authenticate the user and return a user object
+    user = db.authenticate_user(form.get('username'), form.get('password'))
+    session['user'] = user.frontend_data()
```

Comment:
We're not storing a `user` here, we're storing user frontend data, so the field could be named to reflect that like:
```
session['user_frontend_data'] = user.frontend_data()
```
This is a bit confusing in the `get_user()` route above where it looks like you either get a user or user frontend data.

Reply:
true. I'll try to make this more explicit!